backend/api/EventHandlers/MqttEventHandler.cs

```diff
@@ -236,13 +235,13 @@ private static void UpdateHostIfChanged(string host, ref Robot robot, ref List<s
         private static void UpdateCurrentAssetIfChanged(string newCurrentAsset, ref Robot robot,
             ref List<string> updatedFields)
         {
-            if (newCurrentAsset.Equals(robot.CurrentAsset, StringComparison.Ordinal))
+            if ((robot.CurrentAssetDeck.AssetCode != null) && newCurrentAsset.Equals(robot.CurrentAssetDeck.AssetCode, StringComparison.Ordinal))
```

Comment:
You are still getting Dereference of a possible null reference because you are still trying to access robot.CurrentAssetDeck that might be null.

First you need to check that the `robot.CurrentAssetDeck` is not null